va/va.h

```diff
@@ -2087,22 +2087,35 @@ typedef struct _VAEncMiscParameterHRD
 } VAEncMiscParameterHRD;
 
 /**
- * \brief Defines a maximum frame size (in bits).
+ * \brief Profile level maximum frame size (in bits) and 
+ * frame level maximum frame size (in bytes) settings.
  *
- * This misc parameter buffer defines the maximum size of a frame (in
- * bits). The encoder will try to make sure that each frame does not
- * exceed this size. Otherwise, if the frame size exceeds this size,
+ * This misc parameter buffer defines profile level maximum frame size of a frame 
+ * (in bits) and frame level maximum frame size (in bytes) settings. For profile 
+ * level maximum frame size, the encoder will try to make sure that each frame does 
+ * not exceed this size. Otherwise, if the frame size exceeds this size,
  * the \c status flag of #VACodedBufferSegment will contain
- * #VA_CODED_BUF_STATUS_FRAME_SIZE_OVERFLOW.
+ * #VA_CODED_BUF_STATUS_FRAME_SIZE_OVERFLOW. For frame level maximum frame size,
+ * currently only AVC encoder can support this settings in multipak case. If the
+ * frame size exceeds this size, the encoder will do more pak passes to adjust the
+ * QP value to control the frame size.
  */
 typedef struct _VAEncMiscParameterBufferMaxFrameSize {
     /** \brief Type. Shall be set to #VAEncMiscParameterTypeMaxFrameSize. */
     VAEncMiscParameterType      type;
-    /** \brief Maximum size of a frame (in bits). */
+    /** \brief Profile level maximum size of a frame (in bits). */
     uint32_t                max_frame_size;
-
-    /** \brief Reserved bytes for future use, must be zero */
-    uint32_t                va_reserved[VA_PADDING_LOW];
+    /** \brief delta QP list for every pass, currently AVC encoder can support up to 4 passes*/
+    uint8_t                *multipak_delta_qp;
+    /** \brief frame level max frame size (in bytes) for AVC encoder in multi pak case */
+    uint32_t                multipak_max_frame_size;
```

Comment:
two comments:
- is it necessary to define the multipak_max_frame_size for multi pak case? Is it possible to reuse the member max_frame_size? 
- there is different unit size for multipak_max_frame_size(bytes) and max_frame_size(bits). is it OK to use the same unit?

BTW, here  max_frame_size is not the size specified in the spec for Profile/Level. it is configured by user.

Reply:
For the first question, we cannot combine these two type max frame size in one parameter, because the UMD processes these two type max frame size in different ways. 
For the second question, I suggest to change max_frame_size unit from bits to bytes, since AVC FEI and HEVC FEI max frame size parameters unit are bytes as multipak_max_frame_size. But change this unit from bits to bytes will impact the libva interface compatibility.